### PR TITLE
#166928228 Setup webpack for different environment

### PR DESCRIPTION
**Pull request by Niyitangasam**
Description
=======
This PR adds seperate configuration for webpack in production and development environment.

Type of change
=======
N/A

How has it been tested
=======
In the development environment, it can be tested by running ``npm run dev``

In a production environment, it can be tested by running ``npm run build``

It can be deployed  ``npm run start``

Pivotal tracker story ID
=======
[166928228](https://www.pivotaltracker.com/n/projects/2326984/stories/166928228)

### CODE REVIEW

File: webpack/prod.config.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import merge from "webpack-merge";
+import baseConfig from "./base.config.js";
+import path from "path";
+
+export default merge(baseConfig, {
+  devtool: "source-map",
```
Comment (macphilips) [inline webpack/prod.config.js]: Is this necessary for prod?

File: webpack/prod.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  output: {
+    path: path.join(__dirname, "../build"),
+    filename: "index.bundle.js"
+  },
+  mode: process.env.NODE_ENV || "production",
```
Comment (macphilips) [inline webpack/prod.config.js]:
```suggestion
  mode: "production",
```

File: package.json [hunk 3]
```diff
@@ -5,8 +5,9 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "webpack": "babel-node ./node_modules/webpack/bin/webpack",
-    "start": "babel-node ./node_modules/webpack-dev-server/bin/webpack-dev-server --open",
+    "build": "NODE_ENV=production babel-node ./node_modules/webpack/bin/webpack --config webpack/prod.config.js",
+    "dev": "NODE_ENV=development babel-node ./node_modules/webpack-dev-server/bin/webpack-dev-server --config webpack/dev.config.js",
+    "start": "npm run build && babel-node  server.js",
```
Comment (macphilips) [inline package.json]: Keep the previous start script, so we could use webpack-dev-server for development.
And use `start:prod` for the production build.

**Review by jabichris — APPROVED** [review]

**Review by AJAkimana — APPROVED** [review]
Good job @Niyitangasam

**Pull request merged**